### PR TITLE
Fix nms layer in mixed precision

### PR DESCRIPTION
**Pull request by elad-c**
This PR fixes an issue with tf.image.combined_non_max_suppression. the output of that layer is neither a Tensor or a list of Tensors, which fails the MCT in MixedPrecision. The PR excludes this layer from the MixedPrecision interesrt points.

### CODE REVIEW

File: tests/keras_tests/feature_networks_tests/feature_networks/weights_mixed_precision_tests.py [hunk 1]
```diff
@@ -221,6 +223,54 @@ def compare(self, quantized_model, float_model, input_x=None, quantization_info=
...
+        # kpi is for 4 bits on average
+        return KPI(17920 * 4 / 8)
+
+    def create_networks(self):
+        inputs = layers.Input(shape=self.get_input_shapes()[0][1:])
+        x = layers.Conv2D(32, 4)(inputs)
+        x = layers.BatchNormalization()(x)
+        x = layers.Conv2D(32, 4)(x)
+        x = layers.ReLU()(x)
+        if self.layer_type == 'combined_nms':
```
Comment (ofirgo) [inline tests/keras_tests/feature_networks_tests/feature_networks/weights_mixed_precision_tests.py:242]: maybe create two tests one for each of the options? (more "unit-testy")

File: tests/keras_tests/feature_networks_tests/test_features_runner.py [hunk 2]
```diff
@@ -206,6 +206,10 @@ def test_mixed_precision_search_kpi_2bits_avg(self):
     def test_mixed_precision_search_kpi_4bits_avg(self):
         MixedPercisionSearchKPI4BitsAvgTest(self).run_test()
 
+    def test_mixed_precision_search_kpi_4bits_avg_nms(self):
```
Comment (ofirgo) [inline tests/keras_tests/feature_networks_tests/test_features_runner.py:209]: As we discussed offline - extract to a dedicated custom layers tests package

File: tests/keras_tests/layer_tests/base_keras_layer_test.py [hunk 3]
```diff
@@ -105,15 +106,16 @@ def create_networks(self):
         networks = []
         for i, layer in enumerate(layers):
             inputs = [Input(shape=s[1:]) for s in self.get_input_shapes()]
-            if self.is_inputs_a_list:
+            if isinstance(layer, partial) and layer.func is tf.image.combined_non_max_suppression:
```
Comment (ofirgo) [inline tests/keras_tests/layer_tests/base_keras_layer_test.py:109]: What's the meaning of checking "isinstance(layer, partial)"? why would it be of type "partial"? (If this is something that is sewed for the test then at least add a comment explaining it, or maybe try to find a way to make the flow more explicit)

File: tests/keras_tests/layer_tests/base_keras_layer_test.py [hunk 4]
```diff
@@ -168,12 +170,12 @@ def __compare_float_mode(self, float_model, quantized_model):
         input_tensors = self.generate_inputs()
         y = self.predict(float_model, input_tensors)
         y_hat = self.predict(quantized_model, input_tensors)
-        if isinstance(y, list):
+        if isinstance(y, list) or (isinstance(self.layers[0], partial) and
```
Comment (ofirgo) [inline tests/keras_tests/layer_tests/base_keras_layer_test.py:173]: same

File: tests/keras_tests/layer_tests/test_layers_runner.py [hunk 5]
```diff
@@ -285,6 +285,14 @@ def test_averagepooling2d(self):
                             AveragePooling2D(strides=2),
                             AveragePooling2D(strides=(2, 1))]).run_test()
 
+    def test_combinednms(self):
```
Comment (ofirgo) [inline tests/keras_tests/layer_tests/test_layers_runner.py:288]: Same regarding exporting the tests to a separate package

**Review by ofirgo — CHANGES_REQUESTED** [review]
In addition, maybe add to the PR commit a sentence regarding the new test package

File: tests/keras_tests/feature_networks_tests/feature_networks/weights_mixed_precision_tests.py (hunk 1 above)
Comment (elad-c) [reply tests/keras_tests/feature_networks_tests/feature_networks/weights_mixed_precision_tests.py]: separated the tests

File: tests/keras_tests/feature_networks_tests/feature_networks/weights_mixed_precision_tests.py [hunk 6]
```diff
@@ -221,6 +223,54 @@ def compare(self, quantized_model, float_model, input_x=None, quantization_info=
...
 
+class MixedPercisionSearchKPI4BitsAvgTestCombinedNMS(MixedPercisionBaseTest):
+    def __init__(self, unit_test, layer_type):
+        assert layer_type in ['combined_nms', 'SSDPostProcess']
+        self.layer_type = layer_type
+        super().__init__(unit_test)
+
+    def get_kpi(self):
+        # kpi is for 4 bits on average
+        return KPI(17920 * 4 / 8)
```
[the remark this answers is not in the document]
Comment (elad-c) [reply tests/keras_tests/feature_networks_tests/feature_networks/weights_mixed_precision_tests.py:231]: it's the same test that uses it

File: tests/keras_tests/layer_tests/base_keras_layer_test.py (hunk 3 above)
Comment (elad-c) [reply tests/keras_tests/layer_tests/base_keras_layer_test.py:109]: it's the standard way to mimic tf functions as Keras layers. used extensively in the Keras layer tests.

File: tests/keras_tests/layer_tests/test_layers_runner.py (hunk 5 above)
Comment (elad-c) [reply tests/keras_tests/layer_tests/test_layers_runner.py:288]: A separate package was opened for custom layers. This a native TF layer

File: tests/keras_tests/feature_networks_tests/feature_networks/weights_mixed_precision_tests.py [hunk 7]
```diff
@@ -48,7 +49,8 @@ def get_quantization_config(self):
                                       activation_channel_equalization=True)
 
     def get_mixed_precision_v2_config(self):
-        return mct.core.MixedPrecisionQuantizationConfigV2(num_of_images=1)
+        return mct.core.MixedPrecisionQuantizationConfigV2(num_of_images=1,
+                                                           use_grad_based_weights=False)
```
[the remark this answers is not in the document]
Comment (elad-c) [reply tests/keras_tests/feature_networks_tests/feature_networks/weights_mixed_precision_tests.py:52]: mistake. will only disable it for NMS test

File: tests/keras_tests/layer_tests/base_keras_layer_test.py (hunk 3 above)
Comment (ofirgo) [reply tests/keras_tests/layer_tests/base_keras_layer_test.py:109]: The usage of partial is fine, I was referring for the way that you identify the layer in the test by using `isinstance(..., partial)`, which seems cryptic and unclear.
So, either identify the layer in a different way or at least add a comment explaining what you are expecting to find here

**Review by ofirgo — APPROVED** [review]

**Pull request merged**